ibis/tests/all/test_window.py

```diff
@@ -298,3 +305,44 @@ def test_bounded_preceding_windows(backend, alltypes, df, con, window_fn):
     left, right = result.val, expected.val
 
     backend.assert_series_equal(left, right)
+
+
+@pytest.mark.parametrize(
+    ('result_fn', 'expected_fn'),
+    [
+        param(
+            lambda t, win: t.double_col.mean().over(win),
+            lambda gb: (gb.double_col.transform('mean')),
```

Comment:
Why is this `transform` instead of `agg`?

Reply:
Oh, I see, this is unbound window

Reply:
Is there a test that covers rolling window + udf?

Reply:
I haven't added UDF test parameter to the non-unbounded window tests because PySpark 2 only supports Pandas UDFs with unbounded windows

Although, considering that #2364 is happening in parallel to this, I started looking into whether I can add a check for the PySpark version and only run those test parameters if on PySpark 3

But since these are tests in `ibis/tests/all`, it gets quite complicated. _All_ the CI jobs (for each backend) will use these tests, but not all will have `pyspark` installed in its environment. Possibly would need to have an `import pyspark` in a `try-except`, and then for each UDF test parameter on bounded-window tests, add a check to see if `pyspark` exists and `pyspark.__version__ <  '3.0.0'` (and skip the test parameter if true)

The imports and checks would maybe get a bit cleaner if we copied these tests over to `ibis/pyspark/tests`, but then we'd have duplicated tests. I think I would prefer the other way better (keeping things in `ibis/tests/all`).

What do you think @icexelloss?

Reply:
Yeah if too complicated we can do this as follow up.